### ISSUE TITLE
network_monitor broken in linux

### ISSUE THREAD

**Issue by tbearden** [OpenRCE/sulley#39]
When I use network monitor on my kubuntu precise machine, the network_monitor starts and waits for a connection and then exits immediately when it gets one.  Traced this back to line 181 of the script "self.pcap = pcapy.open_live(self.device, -1, 1, 100)".  Trying this just in a shell I get an error  "pcapy.PcapError: eth0: can't allocate oneshot buffer: Cannot allocate memory".  Changing line 181 to "self.pcap = pcapy.open_live(self.device, 1518, 1, 100)" fixes this, but I haven't tested on windows, so I don't know if it breaks anything there.

**Comment by jtpereyda** [comment]
It also only seemed to happen on Linux for me.

@tbearden correctly identified that the second argument to `open_live` (`snaplen`) is the problem. 

From pcapy docs:
> snaplen specifies the maximum number of bytes to capture.

After some digging I found this bug report: https://github.com/the-tcpdump-group/libpcap/issues/99 where someone says:

> Most likely older versions of libpcap treated -1 as a very large unsigned integer and did not detect the memory allocation failure. Now that bug in the library is fixed and the bug in the application is visible better.

#### TL;DR
giving -1 to `open_live`'s second argument is some kind of pcap library hack and not officially supported, and therefore breaks in some instances.

#### Solution
Change the -1 to some large positive integer. The maximum 32 bit integer (2,147,483,647) value should do what -1 was doing before, but lots of people online seem to use 65,535. 2 gigs seems like a lot for a packet anyway.

I'm out of time tonight but maybe I can do this soon. The fix is simple anyway.
